firedrake/preconditioners/pmg.py

```diff
@@ -529,51 +736,53 @@ def make_blas_kernels(self, Vf, Vc):
             return;
         }}
         """
-        self.prolong_kernel = op2.Kernel(prolong_code, "prolongation", ldargs=["-lpetsc"])
-        self.restrict_kernel = op2.Kernel(restrict_code, "restriction", ldargs=["-lpetsc"])
+
+        from firedrake.slate.slac.compiler import BLASLAPACK_LIB, BLASLAPACK_INCLUDE
+        prolong_kernel = op2.Kernel(prolong_code, "prolongation", include_dirs=BLASLAPACK_INCLUDE.split(), ldargs=BLASLAPACK_LIB.split())
+        restrict_kernel = op2.Kernel(restrict_code, "restriction", include_dirs=BLASLAPACK_INCLUDE.split(), ldargs=BLASLAPACK_LIB.split())
+        return prolong_kernel, restrict_kernel
 
     @staticmethod
-    def get_nodes_1d(V):
-        # Return GLL nodes if V==CG or GL nodes if V==DG
-        from FIAT import quadrature
-        from FIAT.reference_element import DefaultLine
+    def get_line_element(V):
```

Comment:
This looks like it's the same as the free function `get_line_element` above.

Reply:
Right, these are duplicates. I'm deleting the static methods from above.